src/ui/modal.svelte

```diff
@@ -1,43 +1,53 @@
 <script lang="ts">
   import { type Snippet } from "svelte";
-
+  import { fade } from "svelte/transition";
   export let style = "";
+  export let as: string = "div";
   export let children: Snippet;
+  export let delayFade = 130;
 </script>
 
-<div class="rcb-modal-container">
-  <main class="rcb-modal-main" {style}>
+<div class="rcb-modal-container" {style}>
+  <svelte:element
+    this={as}
+    class="rcb-modal-main"
+    transition:fade={{ duration: 500, delay: delayFade }}
```

Comment:
opening the modal directly felt a bit off. Adding a slight transition makes it look nice imo